contracts/BalanceTrackerBase.sol

```diff
@@ -310,12 +312,8 @@ abstract contract BalanceTrackerBase {
 
         // Get requester balance
         uint256 requesterBalance = mapRequesterBalances[requester];
-        // Check requester balance
-        if (requesterBalance < totalMechDeliveryRate) {
-            revert InsufficientBalance(requesterBalance, totalMechDeliveryRate);
-        }
         // Adjust requester balance
-        requesterBalance -= totalMechDeliveryRate;
+        requesterBalance = _adjustInitialBalance(requester, requesterBalance, totalMechDeliveryRate, paymentData);
```

Comment:
Was a bug for NVM - corrected to calculate subscription balances correctly when delivering with signatures. Added corresponding tests.